Initial commit.

LittleGizen opened the pull request:
Added structure and files with logic for homework.

Commit:
Initial commit.
Added structure and files with logic for homework.

Alexandr056 commented on src/main/java/ru/innotech/study/TestRunner.java:
```diff
@@ -0,0 +1,91 @@
+     */
+    private void checkByRules(Class c) {
+        List<Annotation> annotationList = new ArrayList<>();
+        Arrays.stream(c.getMethods()).forEach(method -> annotationList.addAll(List.of(method.getDeclaredAnnotations())));
```
создаете стрим (все методы класса, перебираем и собираем в список аннотаций, все аннотации представленные на методе?) причем каждый метод, а аннотации в одну кучу? зачем?

стрим в отдельный объект не выделяете, терминальной операии нет

LittleGizen replied:
Думал, что forEach терминальный. 
Поправил.

Alexandr056 commented on src/main/java/ru/innotech/study/TestRunner.java:
```diff
@@ -0,0 +1,91 @@
+     */
+    private List<Method> getSortedTests(Class c) {
+        List<Method> testsList = new LinkedList<>();
+        Arrays.stream(c.getMethods()).forEach(method -> {
```
еще перебор методов…

LittleGizen replied:
Убрал.

Alexandr056 commented on src/main/java/ru/innotech/study/TestRunner.java:
```diff
@@ -0,0 +1,91 @@
+        Arrays.stream(c.getMethods()).forEach(method -> annotationList.addAll(List.of(method.getDeclaredAnnotations())));
+
+        //Check priority range in Test annotation
+        annotationList.forEach(annotation -> {
```
отдельно проходить все что бы проверить только диапазон значений от 0 до 10 избыточно

LittleGizen replied:
Поправил.

Alexandr056 commented on src/main/java/ru/innotech/study/TestRunner.java:
```diff
@@ -0,0 +1,91 @@
+        });
+
+        List<String> annotationsNamesList = new ArrayList<>();
+        annotationList.forEach(annotation -> {
```
снова перебор аннотаций
собираете имена отдельно?

что-то очень сложная конструкция

LittleGizen replied:
Убрал.

Alexandr056 reviewed (changes requested):
сложная и спорная реализация, давайте проще:
один проход всех методов и валидация аннотаций, раскладываем по кучкам

далее если все ок, то вызываем их

Commit:
The class has been corrected according to the comments:
1. Stream API has been removed
2. The design is simplified
3. Fixed to one pass through methods

LittleGizen commented:
> сложная и спорная реализация, давайте проще: один проход всех методов и валидация аннотаций, раскладываем по кучкам
> 
> далее если все ок, то вызываем их

Чуть упростил конструкцию.

Alexandr056 commented on src/main/java/ru/innotech/study/TestRunner.java:
```diff
@@ -0,0 +1,82 @@
+            AfterSuite.class, 1
+    );
+
+    public void runTests(Class c) throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, InstantiationException {
```
можно схлопнуть до Exception

Alexandr056 commented on src/main/java/ru/innotech/study/TestRunner.java:
```diff
@@ -0,0 +1,82 @@
+    private void checkPriorityRange(Method method) {
+        int priority = method.getDeclaredAnnotation(Test.class).priority();
+        if (priority < 1 || priority > 10) {
+            System.err.printf("Annotation %s priority out of range (1-10) in method %s.\n", Test.class, method);
```
а приложение дальше будет работать? просто напишет что ошибка? 
здесь место для исключения

Alexandr056 commented on src/main/java/ru/innotech/study/TestRunner.java:
```diff
@@ -0,0 +1,82 @@
+
+    private void checkStaticModifier(Method method, Class annotationClass) {
+        if (!Modifier.isStatic(method.getModifiers())) {
+            System.err.printf("Method %s marked with %s but not static.\n", method, annotationClass);
```
аналогично

Alexandr056 reviewed (approved):
хорошая работа над ошибками, есть еще пара комментариев, но работа хорошая и сейчас достаточно для того что бы ее принять. ок
